subprojects/groovy-jsr223/src/main/java/org/codehaus/groovy/jsr223/GroovyScriptEngineImpl.java

```diff
@@ -427,7 +427,7 @@ private Object callGlobal(String name, Object[] args, ScriptContext ctx) {
     }
 
     // generate a unique name for top-level Script classes
-    private synchronized String generateScriptName() {
+    private synchronized static String generateScriptName() {
```

Comment:
I actually thought about this one, as it sure is a slight semantic change here. The method increments a counter variable, that itself is static. I don't thing having that synchronized on a per-object basis does make sense, or does it?